src/main/java/com/iwaconsolti/league/demo/config/LeagueInitializer.java

```diff
@@ -0,0 +1,75 @@
+package com.iwaconsolti.league.demo.config;
+
+import com.iwaconsolti.league.demo.model.*;
+import com.iwaconsolti.league.demo.service.LeagueService;
+import com.iwaconsolti.league.demo.service.TeamService;
+import jakarta.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Profile("populated")
+public class LeagueInitializer {
+    private final LeagueService leagueService;
+    private final TeamService teamService;
+    @Value("${league.max-teams}")
+    private int maxTeams;
+
+    public LeagueInitializer(LeagueService leagueService, TeamService teamService) {
+        this.leagueService = leagueService;
+        this.teamService = teamService;
+        initializeData();
```

Comment:
You are creating the information twice, once with this call to initializeData and other withe the PostConstruct